lib/module.js

```diff
@@ -219,13 +219,11 @@ Module._findPath = function(request, paths, isMain) {
       if (exts === undefined)
         exts = Object.keys(Module._extensions);
       filename = tryPackage(basePath, exts, isMain);
-    }
 
-    if (!filename && rc === 1) {  // Directory.
-      // try it with each of the extensions at "index"
-      if (exts === undefined)
```

Comment:
I think we may need to keep this line and the one below it, since it is used for lazy creation.